### PR TITLE
More SCP Events (049)

### PR DESCRIPTION
**Pull request by xNexus-ACS**

### CODE REVIEW

File: Exiled.Events/Patches/Events/Scp049/Consuming.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    /// <summary>
+    ///     Patches <see>
+    ///         <cref>RagdollAbilityBase{T}.ServerProcessCmd</cref>
+    ///     </see>
+    ///     .
+    ///     Adds the <see cref="Handlers.Scp049.ConsumingCorpse" /> event.
+    /// </summary>
+    [HarmonyPatch]
+    public class Consuming
```
Comment (louis1706) [inline Exiled.Events/Patches/Events/Scp049/Consuming.cs:36]: put //TODO Rework Transpiler

File: Exiled.Events/Patches/Events/Scp049/SendingCall.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    using PlayerRoles.PlayableScps.Scp049;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    ///     Patches <see cref="Scp049CallAbility.ServerProcessCmd" />.
+    ///     Adds the <see cref="Handlers.Scp049.SendingCall" /> event.
+    /// </summary>
+    [HarmonyPatch]
+    public class SendingCall
```
Comment (louis1706) [inline Exiled.Events/Patches/Events/Scp049/SendingCall.cs:28]: put //TODO Rework Transpiler

**Review by louis1706 — APPROVED** [review]

File: Exiled.Events/Patches/Events/Scp049/ActivatingSense.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    using Utils.Networking;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    ///     Patches <see cref="Scp049ResurrectAbility.ServerComplete" />.
+    ///     Adds the <see cref="Handlers.Scp049.ActivatingSense" /> event.
+    /// </summary>
+    [HarmonyPatch]
+    public class ActivatingSense
```
Comment (louis1706) [inline Exiled.Events/Patches/Events/Scp049/ActivatingSense.cs:35]: put //TODO Rework Transpiler

**Review by louis1706 — APPROVED** [review]

File: Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    using API.Features;
+
+    using Interfaces;
+
+    using PlayerRoles.PlayableScps.Scp049.Zombies;
+
+    /// <summary>
+    ///     Contains all information before zombie consumes RagDolls.
+    /// </summary>
+    public class ConsumingCorpseEventArgs : IPlayerEvent, IDeniableEvent
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs:19]: IRagdollEvent

File: Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+        /// <summary>
+        ///     Gets the player who is controlling SCP-049-2.
+        /// </summary>
+        public Player Player { get; }
+
+        /// <summary>
+        ///     Gets the RagDoll to be consumed.
+        /// </summary>
+        public BasicRagdoll Ragdoll { get; }
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs:45]: Use Exiled API :)

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs (hunk 4 above)
Comment (xNexus-ACS) [reply Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs:19]: oh, right, I forgot that exists XD

Comment (xNexus-ACS) [reply Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs]: Done

File: Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs (hunk 5 above)
Comment (xNexus-ACS) [reply Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs]: Done

File: Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+        /// <summary>
+        ///     Gets the RagDoll to be consumed.
+        /// </summary>
+        public Ragdoll Ragdoll { get; }
+
+        /// <summary>
+        /// Gets or sets error code to send back to client.
+        /// </summary>
+        public byte ErrorCode { get; set; }
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs:50]: use Enum

**Review by louis1706 — CHANGES_REQUESTED** [review]
Sorry for this request change again

File: Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        /// </summary>
+        public Ragdoll Ragdoll { get; }
+
+        /// <summary>
+        /// Gets or sets error code to send back to client.
+        /// </summary>
+        public ZombieConsumeAbility.ConsumeError ErrorCode { get; set; }
+
+        /// <summary>
+        ///     Gets or sets a value indicating whether or not the server will send 049 information on the recall.
```
Comment (Thundermaker300) [inline Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs:53]: Wrong documentation

**Review by Thundermaker300 — CHANGES_REQUESTED** [review]

File: Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs (hunk 7 above)
Comment (xNexus-ACS) [reply Exiled.Events/EventArgs/Scp049/ConsumingCorpseEventArgs.cs]: Done

**Review by IRacle1 — APPROVED** [review]

**Review by louis1706 — APPROVED** [review]

**Pull request merged**